### DISCUSSION TITLE
Is it possible to add multiple ChatCompletion (AzureOpenID and VertexAIGemini) in one website?

### DISCUSSION THREAD

**Discussion by mymaher** [microsoft/semantic-kernel#7522] [Q&A] [answered]
Hi Everyone,

I have a requirement where I have to add two services (AzureOpenAI and VertexAIGemini) and call the specific service based on module selection.

In my code, when I tried to get required service instance I am getting casting error. It only return the last service which was added.

Code:
-------------------------------------------------------
string geminiModelId = "gemini-1.5-pro";
string geminiLocation = "-";
string geminiProject = "-";

#pragma warning disable SKEXP0070
var kernelBuilder = builder.Services.AddKernel();
**builder.Services.AddVertexAIGeminiChatCompletion(geminiModelId,
	async () =>{return await GetToken();},
		 geminiLocation, geminiProject, Microsoft.SemanticKernel.Connectors.Google.VertexAIVersion.V1);**

**builder.Services.AddAzureOpenAIChatCompletion("gpt-4o", "https://gpt-04o.openai.azure.com/", Constants.GPT4Key);**

var app = builder.Build();

Kernel kernel = app.Services.GetRequiredService<Kernel>();

**VertexAIGeminiChatCompletionService chat = (VertexAIGeminiChatCompletionService)app.Services.GetRequiredService<IChatCompletionService>();**

Exception:
System.InvalidCastException: 'Unable to cast object of type 'Microsoft.SemanticKernel.Connectors.OpenAI.AzureOpenAIChatCompletionService' to type 'Microsoft.SemanticKernel.Connectors.Google.VertexAIGeminiChatCompletionService'.'
-------------------------------------------------------

I investigated further and found that IChatCompletionService added as Singleton in [OpenAIServiceCollectionExtensions.cs](https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/Connectors/Connectors.OpenAI/OpenAIServiceCollectionExtensions.cs).

Is it possible to support multiple Chat Completion Service (OpenAI, Gemini, etc...) in one API (website) and get the instance based on need?

**Comment by sophialagerkranspandey** [comment, marked as answer]
We'd recommend you can specify the service ID and retrieve the service.

**Reply by mymaher** [reply]
Thank you!
